Source/Internal/UICollectionViewLayout+InteractiveReordering.m

```diff
@@ -0,0 +1,202 @@
+    // invalidationContextForEndingInteractiveMovementOfItemsToFinalIndexPaths:previousIndexPaths:movementCancelled:
+    UICollectionViewLayoutInvalidationContext *originalContext =
+    [self ig_invalidationContextForEndingInteractiveMovementOfItemsToFinalIndexPaths:indexPaths previousIndexPaths:previousIndexPaths movementCancelled:movementCancelled];
+
+    return [self ig_cleanupInvalidationContext:originalContext];
+}
+
+- (UICollectionViewLayoutInvalidationContext *)ig_cleanupInvalidationContext:(UICollectionViewLayoutInvalidationContext *)originalContext {
+    IGListAdapter *adapter = (IGListAdapter *)objc_getAssociatedObject(self, kIGListAdapterKey);
+    if (adapter == nil || !self.collectionView) {
+        return originalContext;
+    }
+
+    NSInteger numSections = [adapter numberOfSectionsInCollectionView:(UICollectionView * _Nonnull)self.collectionView];
+
+    // protect against invalidating an index path that no longer exists
+    // (like item 1 in the last section after interactively reordering an item to the end of a list of 1 item sections)
+    if ([originalContext.invalidatedItemIndexPaths count] > 0) {
+        NSUInteger indexToRemove = NSNotFound;
+
+        indexToRemove = [originalContext.invalidatedItemIndexPaths indexOfObjectPassingTest:
+                         ^BOOL(NSIndexPath * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+                             if (obj.section == numSections-1) {
+                                 IGListSectionController *section = [adapter sectionControllerForSection:obj.section];
+                                 return obj.item > [section numberOfItems]-1;
+                             }
+                             return NO;
+                         }];
+
+        if (indexToRemove != NSNotFound) {
+            NSMutableArray<NSIndexPath *> *invalidatedItemIndexPaths = [originalContext.invalidatedItemIndexPaths mutableCopy];
+            [invalidatedItemIndexPaths removeObjectAtIndex:indexToRemove];
+
+            UICollectionViewLayoutInvalidationContext *modifiedContext;
+            if ([originalContext isKindOfClass:[UICollectionViewFlowLayoutInvalidationContext class]]) {
+                // UICollectionViewFlowLayout has a special invalidation context subclass
+                UICollectionViewFlowLayoutInvalidationContext *flowModifiedContext =
+                [UICollectionViewFlowLayoutInvalidationContext new];
+
+                flowModifiedContext.invalidateFlowLayoutDelegateMetrics =
+                [(UICollectionViewFlowLayoutInvalidationContext *)originalContext invalidateFlowLayoutDelegateMetrics];
+                flowModifiedContext.invalidateFlowLayoutAttributes =
+                [(UICollectionViewFlowLayoutInvalidationContext *)originalContext invalidateFlowLayoutAttributes];
+                modifiedContext = flowModifiedContext;
+            }
+            else {
+                modifiedContext = [UICollectionViewLayoutInvalidationContext new];
```

Comment:
This needs to be:

```objc
modifiedContext = [[self layoutAttributesClass] new];
```

Reply:
I think this was supposed to be `invalidationContextClass`

Reply:
Errr ya, that one